cmake/crypto.cmake

```diff
@@ -28,8 +47,9 @@ endif()
 
 add_library(ccfcrypto.host STATIC ${CCFCRYPTO_SRC})
 add_san(ccfcrypto.host)
-target_compile_options(ccfcrypto.host PRIVATE -stdlib=libc++)
-target_link_libraries(ccfcrypto.host PRIVATE crypto)
+target_compile_options(ccfcrypto.host PUBLIC -stdlib=libc++)
+target_link_options(ccfcrypto.host PUBLIC -stdlib=libc++)
```

Comment:
Maybe we could define a use_libcxx(), like we've done for mbedtls etc.

Reply:
There is `LINK_LIBCXX` in `common.cmake`, should I use that?

Reply:
Yes!